contracts/PWNDeed.sol

```diff
@@ -162,42 +215,106 @@ contract PWNDeed is ERC1155, Ownable {
         bytes memory _signature,
         address _sender
     ) external onlyPWN {
-        bytes32 eip712DomainSeparator = keccak256(abi.encode(
-            keccak256("EIP712Domain(string name,string version,uint256 chainId,address verifyingContract)"),
-            keccak256(bytes("PWN")),
-            keccak256(bytes("1")),
-            block.chainid,
-            address(this)
-        ));
-
         bytes32 offerHash = keccak256(abi.encodePacked(
-            "\x19\x01", eip712DomainSeparator, hash(_offer)
+            "\x19\x01", _eip712DomainSeparator(), hash(_offer)
         ));
 
-        if (_offer.lender.code.length > 0) {
-            require(IERC1271(_offer.lender).isValidSignature(offerHash, _signature) == EIP1271_VALID_SIGNATURE, "Signature on behalf of contract is invalid");
-        } else {
-            require(ECDSA.recover(offerHash, _signature) == _offer.lender, "Lender address didn't sign the offer");
-        }
-        require(_offer.expiration == 0 || block.timestamp < _offer.expiration, "Offer is expired");
-        require(revokedOffers[offerHash] == false, "Offer is revoked or has been accepted");
+        _checkValidSignature(_offer.lender, offerHash, _signature);
+        _checkValidOffer(_offer.expiration, offerHash);
 
         revokedOffers[offerHash] = true;
 
-        id++;
```

Comment:
just curious, what's the reasoning here?

Reply:
It's not necessary but like additional safety net to keep id locally scoped. In case an attacker would re-enter this function, the id for the original call will not increase. In practice, it would only affect id in the DeedCreated event as there is no external call until mint of a deed token.

It make sense to me to have it protected for events even though it's not security issue as we use them to assign deed id to lender.

Reply:
Fair point